### ISSUE TITLE
Use string literal type in `ArrayBuffer`'s `Symbol.toStringTag`

### ISSUE THREAD

**Issue by petamoriken** [microsoft/TypeScript#60149]
### ⚙ Compilation target

ES2023

### ⚙ Library

N/A

### Missing / Incorrect Definition

For ES2023 or lower targets, it is acceptable to assign `SharedArrayBuffer` to an `ArrayBuffer` type. On the other hand, the reverse is not possible because `Symbol.toStringTag` of `SharedArrayBuffer` is a string literal type. 

This problem can be solved by making `Symbol.toStringTag` of `ArrayBuffer` a string literal type. `Set`, `Map`, and `Promise` use the `string` type for library compatibility (#19006), while `ArrayBuffer` is not intended to be extended by the user, and indeed `Symbol.species` support for `InitializeTypedArrayFromTypedArray` has been removed (https://github.com/tc39/ecma262/pull/2719), so there should be no problem.

This issue was reported by #59417 and has been addressed by making TypedArrays generic for the ES2024 target.

### Sample Code

```TypeScript
// ES2023 or lower targets, no type error
const buffer: ArrayBuffer = new SharedArrayBuffer(256);
```

**Comment by petamoriken** [comment]
FYI: [Restricting subclassing support in built-in methods – Data (TC39 meeting slide for 2024-10)](https://docs.google.com/presentation/d/1J0xct8EHUC90P6QpggISxuAp_5L_qaDa3SZ9gQaB0kA/edit#slide=id.p)

**Comment by milkcask** [comment]
Somewhat related: https://github.com/microsoft/TypeScript/pull/60142
Same approach, using `string` instead of the exact well-known tags.

@petamoriken thanks for the FYI !
